types/comma-number/index.d.ts

```diff
@@ -0,0 +1,16 @@
+// Type definitions for comma-number 2.1
+// Project: https://github.com/elidoran/comma-number
+// Definitions by: Corey Rice <https://github.com/coreyar>
+// Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
+
+declare function commaNumber(
+    inputNumber: number | string,
+    optionalSeparator?: string,
+    optionalDecimalChar?: string,
+): string;
+
+declare namespace commaNumber {
+    function bindWith(separator: string, decimalChar: string): (num: number | string) => string;
+}
```

Comment:
you can write this that way (or similar) as the package accepts any type and uses tyepof checks to determine outcome:

```ts

declare function commaNumber<Input>(
    inputNumber: Input,
    optionalSeparator?: string,
    optionalDecimalChar?: string,
): Input extends string | number ? string : Input;

declare namespace commaNumber {
    function bindWith(
        separator: string,
        decimalChar: string,
    ): <Input>(inputNumber: Input) => Input extends string | number ? string : Input;
}
```

```ts
commaNumber('100'); // $ExpectType string
commaNumber('100', '.'); // $ExpectType string
commaNumber(100); // $ExpectType string
commaNumber(false); // $ExpectType false
commaNumber(['a']); // $ExpectType string[]
commaNumber(100, ',');
commaNumber.bindWith(',', '.')(5); // $ExpectType string
commaNumber.bindWith(',', '.')(false); // $ExpectType false
```

Reply:
https://github.com/elidoran/comma-number#version-2-changes
> input with a type other than string and number is returned as is, not as '0'.